packages/asset/contracts/Asset.sol

```diff
@@ -385,6 +301,30 @@ contract Asset is
         return ERC1155URIStorageUpgradeable.uri(tokenId);
     }
 
+    function getTokenIdByMetadataHash(
+        string memory metadataHash
+    ) public view returns (uint256) {
+        return hashUsed[metadataHash];
+    }
+
+    function setMetadataHashUsed(
```

Comment:
I believe setMetadataHashUsed should be called by mint() and batchMint() and be internal

Reply:
That is a good point and I agree.
Commit with the change: https://github.com/thesandboxgame/sandbox-smart-contracts/pull/978/commits/3b47b949f68b2332bf138ca938a156c9479f19c8